prow/jenkins/controller.go

```diff
@@ -480,18 +480,10 @@ func (c *Controller) RunAfterSuccessCanRun(parent, child *kube.ProwJob, ca confi
 		// Not sure what is more correct here. Run the child for now.
 		return true
 	}
-	if ps.RunIfChanged == "" {
-		return true
-	}
-	changesFull, err := ghc.GetPullRequestChanges(org, repo, prNum)
+	shouldRun, err := ps.ShouldRun(parent.Spec.Refs.BaseRef, "", config.NewGitHubDeferredChangedFilesProvider(ghc, org, repo, prNum))
```

Comment:
This isn't technically the same as it didn't check that the job should have run against the branch in the first place or that it set `always_run: true`.

Reply:
Meaning we're slightly changing the logic here? But not changing the end user behavior? SGTM

+1 for standardizing this